news/2021-11-08-performance-points-star-rating-updates.md

```diff
@@ -0,0 +1,132 @@
+
+Consider these two rhythms:
+
+![](/wiki/shared/news/2021-11-08-performance-points-star-rating-updates/rhythm-complexity.png)
+
+Before this difficulty calculation change, the first pattern would have always had greater strain than the second.  In reality, the second can be as demanding or even harder (or potentially harder) than the first depending on the map. With these new changes, the algorithm now more accurately reflects this.
+
+These changes are exceedingly complex from a mathematical point of view. To keep things as simple as possible, notes are grouped up into 'islands', with a triple expressed as a size 3 'island'. Each island is then grouped by the algorithm and considered in a variety of ways, such as:
+
+- islands becoming smaller or larger (size 5 to size 3, for example)
+- islands which change from even to odd sizes
+- the ratio of time elapsed between islands (changes like 1/4 streams into 1/3 streams)
+
+These changes all collectively combine to produce a rhythm value which then adjusts the difficulty of a pattern, from which star rating is ultimately derived.
+
+We hope this change will bring a renewed interest in rhythmically complex songs on osu! and will incentivize players to improve their finger control. 
+
+Good luck setting new, impressive top plays that showcase your skills!
+
+### Total star rating adjustments
+
+In a nutshell, osu! difficulty calculation processes a map and produces skill rating values. These skill values are then manipulated in two different ways to produce two different numbers that you’ll be familiar with: star rating and performance points. 
+
+The correlation between these two numbers has now been improved, following the mathematical derivation work from [**Naitsirk**](https://osu.ppy.sh/users/8202998) in [this pull request](https://github.com/ppy/osu/pull/13986). 
+
+To summarize these changes briefly, maps which focus on one skill will now receive lower star rating than ones which focus on both. **These changes do not affect performance points.**
+
+Here are a few examples:
+
+- [DragonForce - Through the Fire and Flames (Ponoyoshi) [Myth]](https://osu.ppy.sh/beatmapsets/382400#osu/1001682) 6.1* -> 6.32*
+- [Panda Eyes & Teminite - Immortal Flame (feat. Anna Yvette) (Sotarks) [Eternal Blaze]](https://osu.ppy.sh/beatmapsets/703957#osu/1489207) 7.05* -> 6.94*
+
+As for the formula change itself, you can compare and contrast between these:
+
+Old formula: ![](/wiki/shared/news/2021-11-08-performance-points-star-rating-updates/old-sr-white.png)
+New formula: ![](/wiki/shared/news/2021-11-08-performance-points-star-rating-updates/new-sr-white.png)
+
+Interested in what each term means? [**This document**](https://docs.google.com/document/d/10DZGYYSsT_yjz2Mtp6yIJld0Rqx4E-vVHupCqiM4TNI/edit?usp=sharing) written up by [Naitsirk](https://osu.ppy.sh/users/8202998) goes into the nitty gritty details regarding the terms and the derivation itself, for those more mathematically inclined.
```

Comment:
Also found this weird. If the formulas can be replaced with a layman's version that the average user might be able to parse then that would be best.

Reply:
Can only agree with the predecessors. I'd agree with snipping the formulas entirely. This plot from the google doc may get the point across better, I think (but only with explanation as to what is being shown):

![2021-11-08-082811_808x563_scrot](https://user-images.githubusercontent.com/20418176/140700850-ef05759c-be6e-404b-b27c-20dbb7cfbc0b.png)

I can theme it better using my jupyter notebooks I have stashed from the past newsposts if you think that'll work ok.

Reply:
can try it if you want, if you think it's descriptive enough?

Reply:
Yeah I was just about to start preparing something to resolve my unresolved reviews. Sorry for the wait.

Reply:
Ok, so after some sanity loss due to failing to algebra properly, here's the best graph I can give for use instead of the formulas above. I'd just replace them wholesale with this graph.

![sr_pp_correlation](https://user-images.githubusercontent.com/20418176/140804434-87d9edb3-6c55-409b-a683-fd6408ea2cc5.png)

So the way this graph works, is that the axes are ratings for individual skills. The lines represent the combinations of aim & speed skills that a map needs to have to get a final rating of 7*. I don't know whether this is understandable, but I have no idea as to what could be better. Please advise whether you would like me to expand the description further.

Unfortunately I don't think using the ones from the original doc is viable as they are kind of confusing and arbitrary (the axes there don't really represent anything tangible). So I don't have any good idea how to graphically present the "more closely related to pp" part well.

I'd say the link to the doc can stay if someone _really_ wants to look at the mathematical derivation properly.

Reply:
New graph needs to be added to the article.

Reply:
I think this does the trick far better than the formula ever did.